src/foam/java/Constant.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.java',
+  name: 'Constant',
+
+  properties: [
+    'name',
+    'javaType',
```

Comment:
type is understandable.

as for javaValue/value, how would we be able to discern if they have their own custom instantiation or they are simply using a primitive?